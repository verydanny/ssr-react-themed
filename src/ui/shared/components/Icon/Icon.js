import React, { PureComponent } from 'react'
import { themed } from 'react-themed-too'
import cns from 'classnames'
import PropTypes from 'prop-types'

@themed(/^Icon/, { pure: true })
class Icon extends PureComponent {
  static propTypes = {
    name: PropTypes.string,
    title: PropTypes.string,
    className: PropTypes.string,
    onClick: PropTypes.func,
    theme: PropTypes.object,
    svgs: PropTypes.object.isRequired,
  }

  static defaultProps = {
    theme: {},
    svgs: {},
  }

  render() {
    const {
      name,
      className,
      theme,
      svgs,
      ...props
    } = this.props

    const svgKeys = Object.keys(svgs)
    const singleIcon = svgKeys.length === 1
    const svgName = singleIcon ? svgKeys[0] : name
    const Svg = svgs[svgName]

    if (!Svg) return null

    return (
      <span className={cns(className, theme.Icon)} {...props}>
        <Svg {...props} />
      </span>
    )
  }
}

export default Icon
