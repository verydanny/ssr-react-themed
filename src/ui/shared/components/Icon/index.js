export Icon from './Icon'

// UNUSED ICONS // Please only export ones that will actually be used.
// export amenities from './svgs/amenities/amenities.svg'
// export bathroom from './svgs/amenities/bathroom.svg'
// export bedroom from './svgs/amenities/bedroom.svg'
// export apartment from './svgs/apartment.svg'
// export avatar from './svgs/avatar.svg'
// export arrowDown from './svgs/arrowDown.svg'
// export arrowUp from './svgs/arrowUp.svg'
// export back from './svgs/back.svg'
// export bicycle from './svgs/bicycle.svg'
// export car from './svgs/car.svg'
// export check from './svgs/check.svg'
// export crime from './svgs/crime.svg'
// export favorite from './svgs/favorite.svg'
// export unfavorite from './svgs/unfavorite.svg'
// export flag from './svgs/flag.svg'
// export floorplan from './svgs/floorplan.svg'
// export fullscreen from './svgs/fullscreen.svg'
// export fullscreenExit from './svgs/fullscreenExit.svg'
// export heart from './svgs/heart.svg'
// export heartEmpty from './svgs/heartEmpty.svg'
// export hood from './svgs/hood.svg'
// export house from './svgs/house.svg'
// export park from './svgs/park.svg'
// export pencil from './svgs/pencil.svg'
// export phone from './svgs/phone.svg'
// export phoneOutline from './svgs/phoneOutline.svg'
// export refresh from './svgs/refresh.svg'
// export schoolBook from './svgs/schoolBook.svg'
// export schools from './svgs/schools.svg'
// export transport from './svgs/transport.svg'
// export trash from './svgs/trash.svg'
// export vert from './svgs/vert.svg'
// export view from './svgs/view.svg'
// export walk from './svgs/walk.svg'
// export forward from './svgs/forward.svg'
