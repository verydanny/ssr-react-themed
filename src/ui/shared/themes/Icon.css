@value ( RentBlue) from 'ui/shared/themes/colors/colors.css';

.Icon {
  display: inline-block;
  vertical-align: middle;
}

.Icon:after {
  content: "";
  display: table;
  clear: both;
}

/*
NOTE: avoid adding svg dimensions here,
its preferred that they are added via page or device level
themes due to problems with css specificity and the
fact that this file is shared for both large and small icons.
if you need to add share icon dimensions at the device level,
please use a large/Icon.css or small/Icon.css theme
*/
