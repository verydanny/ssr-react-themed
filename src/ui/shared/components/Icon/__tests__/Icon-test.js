import React from 'react'
import { shallow } from 'enzyme'
import Icon from '../Icon'

describe('ui/shared/Icon', () => {
  it('renders with passed svgs object, single item, and no name', () => {
    const props = {
      svgs: { map: 'svg' },
      title: 'bar',
      className: 'logo',
      onClick: jest.fn(),
    }
    const wrapper = shallow(<Icon {...props} />)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with passed svgs object, single item, and mismatched name', () => {
    const props = {
      name: 'list',
      svgs: { map: 'map' },
      title: 'bar',
      className: 'logo',
      onClick: jest.fn(),
    }
    const wrapper = shallow(<Icon {...props} />)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders named svgs with passed svgs object, multiple items', () => {
    const props = {
      name: 'list',
      svgs: { map: 'svg', list: 'list' },
      title: 'bar',
      className: 'logo',
      onClick: jest.fn(),
    }
    const wrapper = shallow(<Icon {...props} />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
