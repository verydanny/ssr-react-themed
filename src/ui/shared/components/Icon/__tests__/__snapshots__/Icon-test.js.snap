// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/shared/Icon renders named svgs with passed svgs object, multiple items 1`] = `

<span class="logo"
      title="bar"
>
  <list title="bar">
  </list>
</span>

`;

exports[`ui/shared/Icon renders with passed svgs object, single item, and mismatched name 1`] = `

<span class="logo"
      title="bar"
>
  <map title="bar">
  </map>
</span>

`;

exports[`ui/shared/Icon renders with passed svgs object, single item, and no name 1`] = `

<span class="logo"
      title="bar"
>
  <svg title="bar">
  </svg>
</span>

`;
